### PR TITLE
Add continuous scroll to Top Instructions

### PR DESCRIPTION
**Pull request by Hamms**
When the scroll buttons are held down for half a second, they now begin
to gradually continuously scroll the target element

scroll

### CODE REVIEW

File: apps/src/templates/instructions/ScrollButtons.jsx [hunk 1]
```diff
@@ -51,6 +53,49 @@ const ScrollButtons = React.createClass({
     return scrollButtonsHeight + (MARGIN * 2);
   },
 
+  componentWillMount: function () {
```
Comment (breville) [inline apps/src/templates/instructions/ScrollButtons.jsx]: possible to just have `componentWillMount() {`?

File: apps/src/templates/instructions/ScrollButtons.jsx [hunk 2]
```diff
@@ -51,6 +53,49 @@ const ScrollButtons = React.createClass({
...
+    if (scrollingUp) {
+      initialScroll *= -1;
+    }
+    scrollBy(contentContainer, initialScroll);
+
+    // If mouse is held down for half a second, begin gradual continuous
+    // scroll
+    var timeout = setTimeout(function () {
+      var interval = setInterval(function () {
+        scrollBy(contentContainer, scrollingUp ? -2 : 2, false);
```
Comment (breville) [inline apps/src/templates/instructions/ScrollButtons.jsx]: can we extract these magic numbers into constants up top, too?

File: apps/src/templates/instructions/ScrollButtons.jsx [hunk 3]
```diff
@@ -51,6 +53,49 @@ const ScrollButtons = React.createClass({
...
+      }.bind(this), 10);
```
Comment (breville) [inline apps/src/templates/instructions/ScrollButtons.jsx]: same comment for the setInterval/setTimeout values.

File: apps/src/templates/instructions/ScrollButtons.jsx [hunk 4]
```diff
@@ -75,12 +120,12 @@ const ScrollButtons = React.createClass({
       <div style={this.props.style}>
         <div
           ref="scrollUp"
-          onClick={this.props.onScrollUp}
+          onMouseDown={this.scrollStart.bind(this, true)}
```
Comment (breville) [inline apps/src/templates/instructions/ScrollButtons.jsx]: nit: true/false for up/down is probably okay, but a named parameter might be a bit more readable.  I had to go to the function to see what true/false meant.

File: apps/src/templates/instructions/ScrollButtons.jsx [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
 import Radium from 'radium';
 import color from '../../color';
-import { getOuterHeight } from './utils';
+import { getOuterHeight, scrollBy } from './utils';
+import { addMouseUpTouchEvent } from '../../dom';
 
 const WIDTH = 20;
 const HEIGHT = WIDTH;
 
+const SCROLL_BY = 100;
```
Comment (breville) [inline apps/src/templates/instructions/ScrollButtons.jsx:11]: maybe some comments on what this and other values mean.. pixels?  when/how often is it called?  etc.

**Comment by breville** [comment]
some comments, but overall lgtm.

circleci seems to have some issues to check.

**Comment by Hamms** [comment]
[Circle failure](https://cucumber-logs.s3.amazonaws.com/circle/24649/ChromeLatestWin7_applab_templateBacked_output.html?versionId=ZDK0omJsm4jCVIghgJnRWWf3q7JmasIo) is unrelated

**Pull request merged**